auth/auth.go

```diff
@@ -1,19 +1,26 @@
 package auth
 
 import (
+	"net/http"
+
 	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/golang/glog"
 	"github.com/rancherio/host-api/app/common"
 	"github.com/rancherio/host-api/config"
-	"net/http"
 )
 
 func Auth(rw http.ResponseWriter, req *http.Request) bool {
 	if !config.Config.Auth {
 		return true
 	}
+	tokenString := req.URL.Query().Get("token")
+
+	if len(tokenString) == 0 {
+		return false
+	}
 
-	token, err := jwt.ParseFromRequest(req, func(token *jwt.Token) (interface{}, error) {
-		return config.Config.Key, nil
+	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+		return config.Config.ParsedPublicKey, nil
```

Comment:
This still does nothing. Please read https://github.com/dgrijalva/jwt-go/blob/master/jwt.go#L135 You need to compare the keys. They are never compared